src/interface/swarm_container.hpp

```diff
@@ -0,0 +1,167 @@
+///
+///
+/// The SwarmContainer class is a container for the swarms of particles that
+/// make up the simulation.
+///
+/// The container class will provide the following methods:
+///
+
+class MeshBlock;
+
+class SwarmContainer {
+ public:
+  //-----------------
+  // Public Variables
+  //-----------------
+  MeshBlock *pmy_block = nullptr; // ptr to MeshBlock
+
+  //-----------------
+  // Public Methods
+  //-----------------
+  // Constructor does nothing
+  SwarmContainer() {}
+
+  ///
+  /// Set the pointer to the mesh block for this container
+  void setBlock(MeshBlock *pmb) { pmy_block = pmb; }
+
+  // TODO BRR also add Add() functions for setting single int, real, string
+  // values?
+
+  ///
+  /// Allocate and add a variable<T> to the container
+  ///
+  /// This function will eventually look at the metadata flags to
+  /// identify the size of the first dimension based on the
+  /// topological location.  Dimensions will be taken from the metadata.
+  ///
+  /// @param label the name of the variable
+  /// @param metadata the metadata associated with the variable
+  ///
+  void Add(const std::string label, const Metadata &metadata);
+
+  ///
+  /// Allocate and add a variable<T> to the container
+  ///
+  /// This function will eventually look at the metadata flags to
+  /// identify the size of the first dimension based on the
+  /// topological location.  Dimensions will be taken from the metadata.
+  ///
+  /// @param labelArray the array of names of variables
+  /// @param metadata the metadata associated with the variable
+  ///
+  void Add(const std::vector<std::string> labelArray, const Metadata &metadata);
+
+  void Add(std::shared_ptr<Swarm> swarm) {
+    swarmVector_.push_back(swarm);
+    swarmMap_[swarm->label()] = swarm;
+  }
+
+  ///
+  /// Get a swarm from the container
+  /// @param label the name of the swarm
+  /// @return the Swarm if found or throw exception
+  std::shared_ptr<Swarm> &Get(std::string label) {
+    if (swarmMap_.count(label) == 0) {
+      throw std::invalid_argument(std::string("\n") + std::string(label) +
+                                  std::string(" swarm not found in Get()\n"));
+    }
+    return swarmMap_[label];
+  }
+
+  std::shared_ptr<Swarm> &Get(const int index) { return swarmVector_[index]; }
+
+  int Index(const std::string &label) {
+    for (int i = 0; i < swarmVector_.size(); i++) {
+      if (!swarmVector_[i]->label().compare(label)) return i;
+    }
+    return -1;
+  }
+
+  /// Gets an array of real variables from container.
+  /// @param names is the variables we want
+  /// @param indexCount a map of names to std::pair<index,count> for each name
+  /// @param sparse_ids if specified is list of sparse ids we are interested in.  Note
+  ///        that non-sparse variables specified are aliased in as is.
+  int GetSwarms(const std::vector<std::string> &names, std::vector<Swarm> &sRet,
+                std::map<std::string, std::pair<int, int>> &indexCount);
```

Comment:
This is missing a function body. It can probably be made const, or it should simply be removed if it is not going to be implemented or used.

Reply:
Removed